Assets/Scripts/Combat/Abilities/MagicShield.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class MagicShield : Ability
+{
+    private enum Button { Up, Down, Left, Right };
+    private enum Phase { SequencePhase, InputSequence, Activate, Inactive };
+    private Timer SequenceTimer;
+    private Queue<Button> Sequence = new Queue<Button>();
+    private Queue<Button> SequenceOrder = new Queue<Button>();
+    private Queue<GameObject> Directions = new Queue<GameObject>();
+
+    [SerializeField]
+    private GameObject Shield;
+    [SerializeField]
+    private GameObject[] Arrows; // 0: Up, 1: Down, 2: Left, 3: Right 
+
+    private readonly float SequenceDuration = 2f;
+    private readonly float InputDuration = 5f;
+    private readonly float PositionOffset = 4f;
+    private Vector3 Position = Vector3.zero;
+    private Phase CurrentPhase = Phase.Inactive;
+
+    private int CorrectInputs = 0;
+    private readonly int MagicShieldFactor = 20;
+    private int MagicShieldHealth;
+
+    public new void Start()
+    {
+        base.Start();
+        SequenceTimer = GetComponent<Timer>();
+        Shield.SetActive(false);
+
+        foreach(GameObject arrow in Arrows)
+            arrow.SetActive(false);
+
+        TargetSchema = new TargetSchema(
+            0,
+            CombatantType.Ally,
+            SelectorType.All);
+    }
+
+    new public void StartAbility()
+    {
+        CorrectInputs = 0;
+        base.StartAbility();
+    }
+
+    private void Update()
+    {
+        if(CurrentPhase == Phase.SequencePhase)
+            SequencePhaseUpdate();
+
+        else if(CurrentPhase == Phase.InputSequence)
```

Comment:
I think it's fine to leave it though, we technically only want one of these options (I'm not sure exactly how this would run, but it may be possible that a second check is always made for the 2nd if, so an else could be a bit of an optimization).

Reply:
Come to think of it, this is a perfect use case for a SWITCH statement. It's OK as is tho, with so few options